uwb_platform_interface/lib/uwb_platform_interface.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+///Contains all the calls that needs to be implemented in the platform specific plugins
+abstract class UwbPlatform extends PlatformInterface {
+  /// Constructs a UwbPlatform.
+  UwbPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  /// The instance of [UwbPlatform] to use.
+  /// Must be set before accessing.
+  static UwbPlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [UwbPlatform] when they register themselves.
+  /// https://github.com/flutter/flutter/issues/43368
+  static set instance(UwbPlatform instance) {
+    PlatformInterface.verify(instance, _token);
+    _instance = instance;
+  }
+
+  /// Should only be accessed after setter is called.
+  static late UwbPlatform _instance;
+
+  /// Must be called once for initial setUp
+  Future<bool?> setUp() =>
+      throw UnimplementedError('setUp() has not been implemented.');
+
+  /// Starts the advertiser
```

Comment:
This is a bit short. Developers new to the ultra wide band topic might not understand what "Start the advertiser" entails. Also the parameters should be explained. A better (more explanatory message) would be:

```suggestion
  /// Start advertising as a utlra wide band host allowing clients to connect.
  ///
  /// The [peerID] identifies a peer (the host in this case) in a  multi peer session.
  /// The [serviceType] describes the service to advertise. This should be a short text string that describes the app's networking protocol.
```